modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/RasterLayerResponse.java

```diff
@@ -259,410 +252,11 @@ public ColorInterpretation getColorInterpretation() {
-            Geometry mask = JTS.toGeometry(new Envelope(bounds.getMinX(), bounds.getMaxX(),
-                    bounds.getMinY(), bounds.getMaxY()));
-            ROI imageROI = new ROIGeometry(mask);
-            if (footprintBehavior.handleFootprints()) {
-
-                // get the real footprint
-                final ROI footprint = result.getFootprint();
-                if (footprint != null) {
-                    if (imageROI.contains(footprint.getBounds2D().getBounds())) {
-                        imageROI = footprint;
-                    } else {
-                        imageROI = imageROI.intersect(footprint);
-                    }
-                }
-
-                // ARTIFACTS FILTERING
-                if (defaultArtifactsFilterThreshold != Integer.MIN_VALUE
-                        && result.isDoFiltering()) {
-                    int artifactThreshold = defaultArtifactsFilterThreshold;
-                    if (artifactsFilterPTileThreshold != -1) {
-
-                        // Looking for a histogram for that granule in order to
-                        // setup dynamic threshold
-                        if (canonicalPath != null) {
-                            final String path = FilenameUtils.getFullPath(canonicalPath);
-                            final String baseName = FilenameUtils.getBaseName(canonicalPath);
-                            final String histogramPath = path + baseName + "." + "histogram";
-                            final Histogram histogram = Utils.getHistogram(histogramPath);
-                            if (histogram != null) {
-                                final double[] p = histogram
-                                        .getPTileThreshold(artifactsFilterPTileThreshold);
-                                artifactThreshold = (int) p[0];
-                            }
-                        }
-                    }
-                    if (LOGGER.isLoggable(Level.FINE)) {
-                        LOGGER.log(Level.FINE, "Filtering granules artifacts");
-                    }
-                    ImageWorker w = new ImageWorker(granule).setRenderingHints(hints)
-                            .setROI(imageROI);
-                    w.setBackground(new double[] { 0 });
-                    w.artifactsFilter(artifactThreshold, 3);
-                    granule = w.getRenderedImage();
-                    // granule = ArtifactsFilterDescriptor.create(granule, imageROI, new double[]{0}, artifactThreshold, 3, hints);
-                }
-            }
-
-            // preparing input
-            return new MosaicElement(alphaChannel, imageROI, granule, result.getPamDataset());
-        }
-    }
-
-    /**
      * A class doing the mosaic operation on top of a List of {@link MosaicElement}s.
-     * 
-     * @author Simone Giannecchini, GeoSolutions SAS
      *
+     * @author Simone Giannecchini, GeoSolutions SAS
      */
-    private class Mosaicker {
+    public static class Mosaicker {
```

Comment:
I can refactor out. I think there are some other inner classes that can be moved to the outer level now.